Operator names

tuschcarsten opened the pull request:
In an old version of the specification, operators were defined by using a special syntax with operator+, operator- etc.
In the current version of the specification, this has been changed to normal method names, using underscores to indicate that those methods are special (etc. _add_). This decision was made by Christopher, Berthold and me, due to the lack of other remaining project members who wanted to work on the specification. The good thing about this is that a special case in the AST generation can be omitted (checking whether the method has an identifier or is an operator).

lummax commented on src/test/resources/testPrograms/expressions/SelfDefinedOperators.output:
```diff
@@ -1 +1 @@
-operator_unary-operator+operator-operator*operator/operator%operator<operator>operator<=operator>=operator=operator!=operator_notoperator_xoroperator_andoperator_or
\ No newline at end of file
+operator_unary-operator+operator-operator*operator/operator%operator<operator>operator<=operator>=operator=operator!=operator_notoperator_andoperator_or
```
Shouldn't you change the output strings aswell? Although it's not necessary.

lummax commented on src/main/java/de/uni/bremen/monty/moco/util/OperatorMapper.java:
```diff
@@ -0,0 +1,125 @@
+
+public class OperatorMapper {
+	public static String methodFromBinaryOperator(String operator) {
+		switch (operator) {
```
Isn't a map way easier?

lummax commented:
If you rebase against moco/master you should be able to loose the merge commit. But it's not important.

tuschcarsten commented:
Tnanks for your reply. I changed the output of the test-case and removed the OperatorMapper class. Instead the ASTBuilder now contains a Map<String, String> for unary and binary operator mappings.

I will rebase this branch if the pull request is accepted (I will then merge it)  in order to not destroy the history of this pull request, but keep the history of the repository clean.
